### PR TITLE
Metrics for Camunda exporter

### LINKED ISSUE #22867: Add metrics for CamundaExporter
- [x] Identify relevant metrics that helps us with performance tuning and debugging 
- [x] Add the metrics to the exporter
- [x] Make the new metrics available in grafana dashboard

### PR DESCRIPTION
**Pull request by EuroLew**
## Description

Micrometer metrics for Camunda exporter, currently tracks
- Failed flushes
- Flush duration
- Flush latency (time from first record in batch to flush command)
- Bulk size (number of entities)
- Bulk size memory

Currently there is no implementation of bulk size memory as the exporter cannot currently track the memory usage of records currently as not implemented.

Screenshots of grafana metrics at https://github.com/camunda/camunda/issues/22867#issuecomment-2397375347

**Success Criteria:**

- [x] Identify relevant metrics that helps us with performance tuning and debugging 
- [x] Add the metrics to the exporter
- [x] Make the new metrics available in grafana dashboard

## Related issues

closes #22867

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/CamundaExporter.java [hunk 1]
```diff
@@ -53,6 +55,15 @@ public class CamundaExporter implements Exporter {
...
   private final ExporterResourceProvider provider;
+  private CamundaExporterMetrics metrics;
+
+  /**
+   * Sample to measure the flush latency of the current bulk request.
+   *
+   * <p>Time of how long an export bulk request is open and collects new records before flushing,
+   * meaning latency until the next flush is done.
+   */
+  private Sample flushLatencyMeasurement;
```
Comment (mustafadagher) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/CamundaExporter.java:66]: 💭   Could we keep track of this instance inside the `CamundaExporterMetrics `?

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/CamundaExporter.java [hunk 2]
```diff
@@ -108,10 +120,21 @@ public void close() {
...
+    if (writer.getBatchSize() == 0) {
+      flushLatencyMeasurement = metrics.startFlushLatencyMeasurement();
+    }
+
     writer.addRecord(record);
     lastPosition = record.getPosition();
+
     if (shouldFlush()) {
-      flush();
+      try (final var ignored = metrics.measureFlushDuration()) {
```
Comment (mustafadagher) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/CamundaExporter.java:131]: 👍 this looked weird at first look, until I went through the micrometer codebase to find that it returns an `AutoCloseable` and only records the duration in the `close()` method

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/metrics/CamundaExporterMetrics.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  public void recordBulkSize(final int bulkSize) {
+    DistributionSummary.builder(meterName("bulk.size"))
+        .description("Exporter bulk size")
+        .serviceLevelObjectives(10, 100, 1_000, 10_000, 100_000)
+        .register(meterRegistry)
+        .record(bulkSize);
+  }
+
+  public void recordBulkMemorySize(final int bulkMemorySize) {
```
Comment (mustafadagher) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/metrics/CamundaExporterMetrics.java:52]: 💭  We are not using this right no (since we actually don't do much yet with the memory size). It doesn't harm to still keep the method until then, so I think it fine

File: monitor/grafana/zeebe.json
Comment (mustafadagher) [inline monitor/grafana/zeebe.json:1]: 💭 I couldn't review the content of this file for good or bad, but I trust you have tested it to work.

**Review by mustafadagher — APPROVED** [review]

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/CamundaExporter.java (hunk 1 above)
Comment (EuroLew) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/CamundaExporter.java:66]: Yes if we do not ever want to track more than 1 instance of flush latency at a time, do you think this will ever occur?

Comment (mustafadagher) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/CamundaExporter.java:66]: I'm not sure I completely understand your point. so let me rephrase

We have one instance of `CamundaExporterMetrics` per instance of `exporter`, right? and basically `flushLatencyMeasurement` is necessarily one per exporter instance too. How is having here different than there? are we sharing this `exporter metrics` instance somehow?

Please elaborate to me if there's something I might be missing 🙏

Comment (EuroLew) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/CamundaExporter.java:66]: I was thinking of a scenario where we would have multiple batch requests (per exporter) at once  (for whatever reason), and we would want to track the latencies of each. But this probably won't happen so I'll move it to metrics
